source/use-cases/sensitive-data-encryption.txt

```diff
@@ -124,7 +124,153 @@ In this step, we generate a local master key to expedite setup of our developmen
           }
 
 
-B. Define a JSON Schema
+B. Create a Data Encryption Key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this section, we use the MongoDB driver to create and store a data
+encryption key, used by CSFLE to automatically encrypt and decrypt data on
+document fields. The following configuration values are required by the client
+to generate a new data encryption key:
+
+* The local master key or AWS KMS master key access settings.
+* A MongoDB connection string that authenticates on a running server.
+* The key vault database and collection name.
+
+Upon creating a data encryption key, the driver returns the ``_id`` field
+(as a `UUID <https://en.wikipedia.org/wiki/Universally_unique_identifier>`_)
+of a document that stores information about the key including:
+
+* The data encryption key, encrypted by the master key.
+* The key management service provider data.
+
+Follow the steps below to generate a single data encryption key from the
+local master key.
+
+.. note::
+
+  If using AWS KMS to provide the master key, refer to the "`F. Convert to
+  a Remote Master Key`_" section for the appropriate settings.
+
+First, read the file containing the local master key generated in the "`A.
+Create a Master Key`_" section:
+
+.. tabs::
+
+  tabs:
+
+    - id: java-master-key-reader
+      name: "Java"
+      content: |
+
+        .. code-block:: java
+
+          String path = "master-key.txt";
+
+          byte[] fileBytes = new byte[96];
+
+          try (FileInputStream fis = new FileInputStream(path)) {
+            fileBytes = fis.readAllBytes();
+          } catch (IOException e) {
+            e.printStackTrace();
+          }
+
+          final byte[] localMasterKey = Arrays.copyOf(fileBytes, 96);
+
+
+Next, specify the KMS provider settings. Since we are using a local master
+key, the provider name is ``local``:
```

Comment:
How do we specify the KMS provider settings? Looks like we make a hashmap in the code but why? What does it contain?

We should add a little bit more here to explain

Reply:
Also these "next..." phrases are really small relative to the code blocks they're sandwiched between. Given that this step is procedural, maybe we can convert to a stepfile?

Reply:
> Looks like we make a hashmap in the code but why? What does it contain?
>
> We should add a little bit more here to explain

Can you explain what you're looking for? Each driver has its own way of specifying settings whether hashmap, json, associative array, etc. Did you want to have explicit information about how drivers specify their configurations? Would that belong in the scope of this document?

Reply:
> Given that this step is procedural, maybe we can convert to a stepfile?

Good idea, will move to a step file.

Reply:
Question about tabs in a step file:
Should the id use the same naming scheme as the document? If so, then they are not reusable in different documents, if there is ever that need. Should we make note of that somewhere?

Reply:
Another question -- if I have a `:ref:` link in the step file, will it resolve to the anchors set in the document that includes the step file?